### PR TITLE
Add new `resizeStyle` to keep aspect ratio and no cropped, fill the rest by background color.

### PR DESCRIPTION
**Pull request by horiuchi**
Could you tell me that the more better name of this resizeStyle.

### CODE REVIEW

File: src/imagemagick.cc [hunk 1]
```diff
@@ -546,6 +566,10 @@ NAN_METHOD(Convert) {
     context->filter = !filterValue->IsUndefined() ?
         *NanAsciiString(filterValue) : "";
 
+    Local<Value> backgroundValue = obj->Get( NanNew<String>("background") );
```
Comment (dustingraves) [inline src/imagemagick.cc]: Change `NanNew<String>("background")` to `Nan::New<String>("background").ToLocalChecked()`

File: src/imagemagick.cc [hunk 2]
```diff
@@ -546,6 +566,10 @@ NAN_METHOD(Convert) {
...
+    context->background = !backgroundValue->IsUndefined() ?
+        *NanAsciiString(backgroundValue) : "";
```
Comment (dustingraves) [inline src/imagemagick.cc]: This becomes `*String::Utf8Value(backgroundValue) : "";`

File: test/test.async.js [hunk 3]
```diff
@@ -17,7 +17,7 @@ function saveToFileIfDebug (buffer, file) {
 test( 'convert filter Lagrange', function (t) {
     t.plan(1);
     imagemagick.convert({
-        srcData: require('fs').readFileSync( "test.png" ), // 58x66
+        srcData: require('fs').readFileSync( "./test/test.png" ), // 58x66
```
Comment (dustingraves) [inline test/test.async.js]: I found that this change broke the tests on my machine

**Comment by dustingraves** [comment]
This works after formatting for Nan 2.x. I forked the master from elad and then retrofitted your changes with the modifications for nan 2.x and was able to make this work. Using the following options: 
```Javascript
{
        srcData: require('fs').readFileSync( "test.png" ), // 58x66
        width: 30,
        height: 100,
        resizeStyle: 'aspectwithbg',
        background: '#ff0000',
        quality: 80,
        format: 'PNG',
        debug: debug
    }
```
out png-aspectwithbg
Adding the path to the tests did not work on my machine.  The background property only works if `resizeStyle` is `aspectwithbg`.

Last thing I would suggest is call it `aspectfitbg` as this better identifies the resize style with added bg prop.

**Comment by dustingraves** [comment]
@horiuchi, I can make a new pull request and update using your changes, or you can make the changes?

@elad how would you like to proceed with this?

**Comment by horiuchi** [comment]
@dustingraves OK, I will send new pull request.
And I close this pull request.

**Pull request closed without merging**